src/sentry/api/endpoints/project_rules.py

```diff
@@ -830,4 +833,28 @@ def post(self, request: Request, project) -> Response:
             wizard_v3=wizard_v3,
         )
 
+        # TODO add msteams and discord
+        SUPPORTED_ACTIONS_MAP = {
+            "sentry.integrations.slack.notify_action.SlackNotifyServiceAction": "sentry.integrations.slack.actions.notification.SlackNotifyServiceAction"
+        }
+
+        # TODO put this in a function that takes 'rule' and 'new' to be shared with project_rule_index
+        for action in rule.data.get("actions", ()):
+            action_cls_id = action.get("id")
+            action_cls = import_string(SUPPORTED_ACTIONS_MAP[action_cls_id])
```

Comment:
Better to use https://github.com/getsentry/sentry/blob/eb55184602940adb1fc09a9c0e33515cac92c7d8/src/sentry/rules/processor.py#L271 to get the class here